### DISCUSSION TITLE
Make an official saver - and recommend it

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6810] [Ideas]
@Jermolene 

There is an [indicated consensus](https://talk.tiddlywiki.org/t/nah-not-for-me-why/3989/17?u=twmat) that tiddly curious people get stuck at the "select a saver" step.

What would it take for tiddlywiki.com to *actively recommend* a specific, and **simple+good**, saving method? While the current "neutral approach" to not recommend anyone is the easy way out, I fear it is not the most productive one.

The... or at least *my* idea... is that we should make this recommendation to *newbies*. They can *change* saver later, but the hurdle is to get started so the simple+good solution is needed at that stage. Just like for any software.

What would it take to "adopt" an unofficial solution to make it official? Or what would it take to create a new one and serve it?

For an online hosted solution I think we should actively promote Tiddlyhost. It is of course already built-in into the $:/ControlPanel so it would make sense to, in GettingStarted on tw.com, write something like "For a simple and free online hosted solution, you can try out Tiddlyhost"

Thanks

**Comment by Jermolene** [comment]
We already have a universal saver: it's the download saver. It's the only one that works everywhere (even iPhones etc).

None of the others are universal in the sense of being available across all platforms, and so I don't see how any of them could be recommended without complex qualifications.

TiddlyWiki is complex. The complexity is what gives rise to the flexibility that users love. The complexity is off-putting for some users. That's a fundamental tension that cannot be resolved by wishing away the complexity.

**Reply by twMat** [reply]
>We already have a universal saver: it's the download saver.

OK, but this is not clear in GettingStarted on tw.com - it is not even mentioned AFAICT, let alone recommended. In GettingStarted it appears that you *must* select one of the savers there.

**Reply by twMat** [reply]
...but, regardless, I doesn't have to be one saver that "works everywhere". It could be a few official savers that together cover the majority of the platforms, as long as they are *official*. It is overwhelming to be presented with a list of badges - what is a visitor to conclude?

**Reply by pmario** [reply]
I think Mat is right here. ... There are many pages, that offer a download button and open the default browser dialogue. For them it's OK that browsers only can save that way and `wiki.html (1)(2)(3)` files are created, if the user uses the default name and clicks OK.

It's only the TW-folk, that insist to use "convenient savers" and since there are so many we constantly get the feedback that it's complicated and confusing. ... 

It isn't ... the "universal saver" works everywhere, we just don't tell anybody.

If we would advertise the default way _only_, create a simple HowTo, I think we would win a lot. ... If users then complain about the `wiki.html (1)(2)(3)` extensions, we know that they "understood" the universal saver. Then we can tell them about the others.

**Comment by Jermolene** [comment]
> OK, but this is not clear in GettingStarted on tw.com - it is not even mentioned AFAICT, let alone recommended. In GettingStarted it appears that you must select one of the savers there.

> ...but, regardless, I doesn't have to be one saver that "works everywhere". It could be a few official savers that together cover the majority of the platforms, as long as they are official. It is overwhelming to be presented with a list of badges - what is a visitor to conclude?

That suggests that we can't promote a single, universal means to save changes, but instead offer a more restricted choice of recommended options: download saver, Node.js, perhaps TiddlyDesktop – suddenly we're back where we started.

Perhaps we could address this by categorising saving techniques according to whether they are (say) "recommended", "stable", or "experimental". We'd make the recommended methods more prominent, and require users to do a bit of clicking before they have to face the full list of experimental saving techniques.

**Comment by twMat** [comment]
>Perhaps we could address this by categorising saving techniques according to whether they are (say) "recommended", "stable", or "experimental". We'd make the recommended methods more prominent, and require users to do a bit of clicking before they have to face the full list of experimental saving techniques.

Yes - and for GettingStarted I would suggest that we *only* include the "recommended" ones.

I wrote the following reply but then realized it was off topic - it isn't really about "making an official saver" but more about *what* to recommend, and to whom, and why. It *is* relevant, especially for my reply above about *only* showing recommended savers in GettingStarted. So somewhat off-topic, but FWIW:

-------

Let's look at who we are talking to and the purpose of this all. My take is this:

We have two demographic groups. People who want a really good saver and people who want to just try out TiddlyWiki. These are also called "experienced users" and "noobs". These are two totally different groups. 

We are not talking about removing the offered savers but rather what the tiddler GettingStarted should present. This tiddler should clearly be for the noobs, hence the tiddler title. **This tiddler should "hook" the visitor enough so that they properly try out TW**. Once someone is down the rabbit hole and they realize they want a more sophisticated saver, well then we can provide that too. But not necessarily in GettingStarted. (It is even an opportunity to promote/teach about the plugins library or links.tiddlywiki.com)

With that said; The demographic group that TW caters to is probably still savvy enough to not be intimidated from a few *very clear* alternatives that they directly understand - for example this group have likely faced "Select Mac or Win32 or Win64 or Linux". These options are mutually exclusive which is particularly desirable. No thinking needed. IMO "single file vs node.js vs TiddlyDesktop" is confusing and requires investigation. GettingStarted is the wrong *time* for that. We should just make it simple for them to *test out TW*. They've never seen a TW beast so *that* is the weirdness they should first get over.

If it was up to me, I'd only show two alternatives in GettingStarted: "Local/downloaded version" VS "Hosted/online version". The former is the native saver. The latter... I wish Unamesa or Jeremy could provide it but the best we have for this is TiddlyHost! Yeah, it doesn't "feel right" to somehow present a local TW file and TH as equals - but imagine if you were a noob who only(!) wanted to try out, say, Evernote and they gave you these two options. Well, it is clear that the hosted option is endorsed and trusted by E.N so you would not care who really owns it. At this stage, you just want a smooth way to try it out, and this is *important*. If E.N is iffy *already at the try out stage* then this really tells you something and there isn't exactly a lack of note-taking alternatives...

I'm sure the people reading this thread use more sophisticated setups than TH, but an end user like me *totally* depend on TH - and I dare say that I am more similar to most in our target group than you are. (But we'll reach the big boys too, don't worry.)

But my main point in this post is that GettingStarted should be about just that. The *multitude* of savers is really for a later step (GettingContinued? GettingOn? GettingFedUp? ;-)

**Comment by pmario** [comment]
@Jermolene 
>...That suggests that we can't promote a single, universal means to save changes, but instead offer a more restricted choice of recommended options: download saver, Node.js, perhaps TiddlyDesktop – suddenly we're back where we started.

Yea, that's probably it. ... As FireFox did "kill" TiddlyFox it seems we where panicking and developed all sorts of "saving alternatives". ... That's a good thing and a bad thing. 

-------

We do have  the Menu-bar now. ... We could make different main menue items, that basically switch between editions and all of those editions have a very similar menu ... So users can switch back and forth between different editions without loosing focus

eg: 

[TiddlyWiki Home](https://tiddlywiki.com) |  [Streams](https://saqimtiaz.github.io/streams/) | [Projectify](https://thaddeusjiang.github.io/Projectify/demo.html) | [GSD5](http://gsd5.tiddlyspot.com/) | [Material-Design](http://j.d.material.tiddlyspot.com/) | [More](https://tiddlywiki.com/#Editions)

**Reply by pmario** [reply]
Material has a "save" button. IMO the TW menu-bar should have that one too

![image](https://user-images.githubusercontent.com/374655/180767244-35f0ceec-3df0-4143-ad6f-db5c696d04e0.png)

**Comment by pmario** [comment]
Edit: 

[TiddlyWiki Home](https://tiddlywiki.com/) | [Streams](https://saqimtiaz.github.io/streams/) | [Projectify](https://thaddeusjiang.github.io/Projectify/demo.html) | [GSD5](http://gsd5.tiddlyspot.com/) | [Material-Design](http://j.d.material.tiddlyspot.com/) | [More](https://tiddlywiki.com/#Editions)

Where More is a dropdown that links to "Community Editions" at tiddlywiki.com ... or something like this

**Comment by Jermolene** [comment]
> We do have the Menu-bar now. ... We could make different main menue items, that basically switch between editions and all of those editions have a very similar menu ... So users can switch back and forth between different editions without loosing focus

I tried that when I was working on the new "sites" menu. My original idea was to have a strip of short names, just as you suggest:

```
docs | dev | links | org 
```

The trouble was that there are too many of them to make it work at mobile screen sizes, hence the dropdown we use at the moment.
